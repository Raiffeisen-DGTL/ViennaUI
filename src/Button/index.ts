export { type ButtonDesign, type ButtonProps, Button } from './Button';
