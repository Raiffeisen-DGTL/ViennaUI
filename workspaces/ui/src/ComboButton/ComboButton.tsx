import React, { useState, useCallback } from 'react';
import { Down, Up } from 'vienna.icons';
import { Wrapper, LeftButton, RightButton } from './ComboButton.styles';
import { DropList } from '../DropList';
import { Props as ItemProps } from '../DropList/Item';
import { ButtonProps } from '../Button';

interface Icons {
    up: React.ReactNode;
    down: React.ReactNode;
}

interface Props extends Pick<ButtonProps, 'design' | 'size'> {
    /** Иконки открытого и закрытого состояния */
    icons?: Icons;

    /** Растягивание опций по ширине родителя */
    fitOptions?: boolean;

    /** Опции */
    options?: React.ReactNode[];

    /** Максимальная высота выпадающего списка в пикселях */
    maxListHeight?: number;
}

const getIcon = (icon, size) => {
    size = size === 'xs' ? 's' : 'm';
    return React.cloneElement(icon, { size });
};

const defaultIcons = { down: <Down key={'1'} />, up: <Up key={'1'} /> };

export const ComboButton: React.FC<Props> & {
    Option: React.FC<ItemProps>;
} = (props: React.PropsWithChildren<Props>) => {
    const { children, fitOptions, options = [], size, design, icons, maxListHeight } = props;

    const [isOpen, setOpen] = useState(false);

    const childrenArr = React.Children.toArray(children);

    const handleClick = useCallback(
        function (this: any, event) {
            if (typeof this.props.onClick === 'function') {
                this.props.onClick(event);
            }
            setOpen(!isOpen);
        },
        [isOpen]
    );

    const handleBlur = useCallback(function (this: any, event) {
        if (typeof this.props.onBlur === 'function') {
            this.onBlur.onClick(event);
        }
        setOpen(false);
    }, []);

    const constructOptions = useCallback(
        () =>
            options.map((option: any) =>
                React.cloneElement(option, {
                    onClick: handleClick.bind(option),
                })
            ),
        [options, handleClick]
    );

    const multipleButttons = React.Children.count(children) > 1;
    const dropbBtn: any = multipleButttons ? childrenArr[1] : childrenArr[0];
    const clickableBtn: any = multipleButttons ? childrenArr[0] : null;

    const dropButtonProps = {
        ...dropbBtn.props,
        square: multipleButttons,
        design,
        size,
        onClick: handleClick.bind(dropbBtn),
        onBlur: handleBlur.bind(dropbBtn),
    };

    const icon = getIcon(icons && (isOpen ? icons.up : icons.down), dropbBtn.props.size);

    return (
        <Wrapper>
            {multipleButttons ? (
                <>
                    <LeftButton {...clickableBtn.props} design={design} size={size} />
                    {React.createElement(RightButton, dropButtonProps, [icon])}
                </>
            ) : (
                React.cloneElement(dropbBtn, dropButtonProps, [dropButtonProps.children, icon])
            )}
            {isOpen && (
                <DropList fitItems={fitOptions} maxHeight={maxListHeight}>
                    {constructOptions()}
                </DropList>
            )}
        </Wrapper>
    );
};

ComboButton.displayName = 'ComboButton';
ComboButton.defaultProps = {
    design: 'primary',
    size: 'm',
    fitOptions: true,
    icons: defaultIcons,
};
ComboButton.Option = DropList.Item;
export default ComboButton;
