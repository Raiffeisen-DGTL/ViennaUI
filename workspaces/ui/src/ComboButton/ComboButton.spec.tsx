import React from 'react';
import { Button, ButtonProps } from '../Button';
import { ComboButton } from './ComboButton';

const options = [
    <ComboButton.Option key={1}>Option 1</ComboButton.Option>,
    <ComboButton.Option key={2}>Option 2</ComboButton.Option>,
    <ComboButton.Option key={3}>Option 3</ComboButton.Option>,
    <ComboButton.Option key={4}>Option 4</ComboButton.Option>,
    <ComboButton.Option key={5}>Option 5</ComboButton.Option>,
];

test('ComboButton', () => {
    const snapOneButton = snapshot.render(
        <ComboButton options={options}>
            <Button>Click Me</Button>
        </ComboButton>
    );

    const snapTwoButton = snapshot.render(
        <ComboButton options={options}>
            <Button>Click Me</Button>
            <Button />
        </ComboButton>
    );

    expect(snapOneButton).toMatchSnapshot();
    expect(snapTwoButton).toMatchSnapshot();
});

test('ComboButton w/ design', () => {
    const designs: ButtonProps['design'][] = ['primary', 'accent', 'critical', 'outline', 'outline-critical', 'ghost'];

    designs.forEach((design) => {
        const snapOneButton = snapshot.render(
            <ComboButton design={design} options={options}>
                <Button>Click Me</Button>
            </ComboButton>
        );

        const snapTwoButton = snapshot.render(
            <ComboButton design={design} options={options}>
                <Button>Click Me</Button>
                <Button />
            </ComboButton>
        );

        expect(snapOneButton).toMatchSnapshot();
        expect(snapTwoButton).toMatchSnapshot();
    });
});

test('ComboButton w/ size', () => {
    const sizes: ButtonProps['size'][] = ['xs', 's', 'm', 'l', 'xl', 'xxl'];

    sizes.forEach((size) => {
        const snapOneButton = snapshot.render(
            <ComboButton size={size} options={options}>
                <Button>Click Me</Button>
            </ComboButton>
        );

        const snapTwoButton = snapshot.render(
            <ComboButton size={size} options={options}>
                <Button>Click Me</Button>
                <Button />
            </ComboButton>
        );

        expect(snapOneButton).toMatchSnapshot();
        expect(snapTwoButton).toMatchSnapshot();
    });
});

test('ComboButton w/ clicks', () => {
    const fn = jest.fn();

    const options = [
        <ComboButton.Option key={1} onClick={fn}>
            Option 1
        </ComboButton.Option>,
    ];

    const fn1 = jest.fn();
    const fn2 = jest.fn();

    const snap = snapshot.shallow(
        <ComboButton options={options}>
            <Button onClick={fn1}>Click Me</Button>
            <Button onClick={fn2} />
        </ComboButton>
    );

    snap.childAt(0).simulate('click', {});
    snap.childAt(1).simulate('click', {});

    expect(fn1).toHaveBeenCalled();
    expect(fn2).toHaveBeenCalled();

    expect(snap.childAt(2).childAt(0).text()).toEqual('Option 1');

    snap.childAt(2).childAt(0).simulate('click', {});
    expect(fn).toHaveBeenCalled();
});

test('ComboButton w/ hide list when blur', () => {
    const options = [<ComboButton.Option key={1}>Option 1</ComboButton.Option>];

    const snap = snapshot.shallow(
        <ComboButton options={options}>
            <Button>Click Me</Button>
            <Button />
        </ComboButton>
    );

    expect(snap.childAt(2).exists()).toEqual(false);

    snap.childAt(1).simulate('click', {});

    expect(snap.childAt(2).exists()).toEqual(true);

    snap.childAt(1).simulate('blur', {});

    expect(snap.childAt(2).exists()).toEqual(false);
});

test('ComboButton w/ hide list when click on item', () => {
    const options = [<ComboButton.Option key={1}>Option 1</ComboButton.Option>];

    const snap = snapshot.shallow(
        <ComboButton options={options}>
            <Button>Click Me</Button>
            <Button />
        </ComboButton>
    );

    snap.childAt(1).simulate('click', {});

    expect(snap.childAt(2).exists()).toEqual(true);

    snap.childAt(2).childAt(0).simulate('click', {});

    expect(snap.childAt(2).exists()).toEqual(false);
});
