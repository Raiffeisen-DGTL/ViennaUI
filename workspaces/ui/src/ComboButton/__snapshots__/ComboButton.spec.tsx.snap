// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComboButton 1`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 dcKhhz"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton 2`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 dcKhhz ComboButtonstyles__LeftButton-jzrgg8-1 difRPs"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
  </button>
  <button
    class="Buttonstyles__Box-qfvnbt-0 hWFVnH ComboButtonstyles__RightButton-jzrgg8-2 erGBcM"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ design 1`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 dcKhhz"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ design 2`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 dcKhhz ComboButtonstyles__LeftButton-jzrgg8-1 difRPs"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
  </button>
  <button
    class="Buttonstyles__Box-qfvnbt-0 hWFVnH ComboButtonstyles__RightButton-jzrgg8-2 erGBcM"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ design 3`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 jolcIP"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ design 4`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 jolcIP ComboButtonstyles__LeftButton-jzrgg8-1 difRPs"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
  </button>
  <button
    class="Buttonstyles__Box-qfvnbt-0 hSbJhv ComboButtonstyles__RightButton-jzrgg8-2 erGBcM"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ design 5`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 gSxLZq"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ design 6`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 gSxLZq ComboButtonstyles__LeftButton-jzrgg8-1 difRPs"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
  </button>
  <button
    class="Buttonstyles__Box-qfvnbt-0 cPykUe ComboButtonstyles__RightButton-jzrgg8-2 erGBcM"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ design 7`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 ebCIkw"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ design 8`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 ebCIkw ComboButtonstyles__LeftButton-jzrgg8-1 difRPs"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
  </button>
  <button
    class="Buttonstyles__Box-qfvnbt-0 gfpLJs ComboButtonstyles__RightButton-jzrgg8-2 fxdXsO"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ design 9`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 lgPlfP"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ design 10`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 lgPlfP ComboButtonstyles__LeftButton-jzrgg8-1 difRPs"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
  </button>
  <button
    class="Buttonstyles__Box-qfvnbt-0 cQyviT ComboButtonstyles__RightButton-jzrgg8-2 erGBcM"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ design 11`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 envJPA"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ design 12`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 envJPA ComboButtonstyles__LeftButton-jzrgg8-1 difRPs"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
  </button>
  <button
    class="Buttonstyles__Box-qfvnbt-0 ePtogg ComboButtonstyles__RightButton-jzrgg8-2 erGBcM"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ size 1`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 gfnTjB"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 gRlLrd"
    >
      Click Me
    </span>
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 gRlLrd"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ size 2`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 gfnTjB ComboButtonstyles__LeftButton-jzrgg8-1 difRPs"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 gRlLrd"
    >
      Click Me
    </span>
  </button>
  <button
    class="Buttonstyles__Box-qfvnbt-0 dxiCgh ComboButtonstyles__RightButton-jzrgg8-2 erGBcM"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 gRlLrd"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ size 3`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 dkSKWW"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ size 4`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 dkSKWW ComboButtonstyles__LeftButton-jzrgg8-1 difRPs"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
  </button>
  <button
    class="Buttonstyles__Box-qfvnbt-0 jxwgnS ComboButtonstyles__RightButton-jzrgg8-2 erGBcM"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ size 5`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 dcKhhz"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ size 6`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 dcKhhz ComboButtonstyles__LeftButton-jzrgg8-1 difRPs"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
  </button>
  <button
    class="Buttonstyles__Box-qfvnbt-0 hWFVnH ComboButtonstyles__RightButton-jzrgg8-2 erGBcM"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ size 7`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 fdUCED"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ size 8`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 fdUCED ComboButtonstyles__LeftButton-jzrgg8-1 difRPs"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
  </button>
  <button
    class="Buttonstyles__Box-qfvnbt-0 etazHX ComboButtonstyles__RightButton-jzrgg8-2 erGBcM"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ size 9`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 kQnzHZ"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ size 10`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 kQnzHZ ComboButtonstyles__LeftButton-jzrgg8-1 difRPs"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      Click Me
    </span>
  </button>
  <button
    class="Buttonstyles__Box-qfvnbt-0 fOQYmN ComboButtonstyles__RightButton-jzrgg8-2 erGBcM"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 kJiCTN"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ size 11`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 KJblR"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 buKajC"
    >
      Click Me
    </span>
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 buKajC"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ComboButton w/ size 12`] = `
<div
  class="ComboButtonstyles__Wrapper-jzrgg8-0 jOwEQG"
>
  <button
    class="Buttonstyles__Box-qfvnbt-0 KJblR ComboButtonstyles__LeftButton-jzrgg8-1 difRPs"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 buKajC"
    >
      Click Me
    </span>
  </button>
  <button
    class="Buttonstyles__Box-qfvnbt-0 eAQYNR ComboButtonstyles__RightButton-jzrgg8-2 erGBcM"
    type="button"
  >
    <span
      class="Buttonstyles__ContentWrapper-qfvnbt-1 buKajC"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
        />
      </svg>
    </span>
  </button>
</div>
`;
